lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -1118,6 +1118,23 @@ LogicalResult ConvertAtenOp<AtenPowTensorTensorOp>::matchAndRewrite(
   return success();
 }
 
+Type getMatMulOutputType(Type inputElemTy, PatternRewriter &rewriter) {
+  Type outputElemTy;
+  if (auto floatTy = dyn_cast<mlir::FloatType>(inputElemTy)) {
+    if (floatTy.isBF16() || floatTy.isF16() || floatTy.isF32()) {
+      // Always accumulate on f32
+      outputElemTy = rewriter.getF32Type();
+    }
+  } else if (auto integerTy = dyn_cast<IntegerType>(inputElemTy)) {
+    if (integerTy.isInteger(/*width=*/8)) {
+      outputElemTy = rewriter.getIntegerType(/*width=*/32);
+    } else if (integerTy.isInteger(/*width=*/16)) {
+      outputElemTy = rewriter.getIntegerType(/*width=*/48);
+    }
+  }
+  return outputElemTy;
```

Comment:
I know that the spec doesn't specify `i32` as input. What happens if we do?
If we get to here, we'd crash because `outputElemTy` wouldn't be defined.

Reply:
No crash will happen, the user of this function checks if the type is valid

Reply:
I stopped here and didn't see where it was used. Sorry for the noise.